From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Victor Stinner <vstinner@python.org>
Date: Mon, 19 Feb 2024 13:20:46 +0100
Subject: [PATCH] 00420: gh-114626: Add again _PyCFunctionFastWithKeywords name
 (GH-115561)

Keep the old private _PyCFunctionFastWithKeywords name (Python 3.7)
as an alias to the new public name PyCFunctionFastWithKeywords
(Python 3.13a4).

_PyCFunctionWithKeywords doesn't exist in Python 3.13a3, whereas
_PyCFunctionFastWithKeywords was removed in Python 3.13a4.
---
 Include/methodobject.h                                        | 2 +-
 .../next/C API/2024-02-16-15-56-53.gh-issue-114626.ie2esA.rst | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)
 create mode 100644 Misc/NEWS.d/next/C API/2024-02-16-15-56-53.gh-issue-114626.ie2esA.rst

diff --git a/Include/methodobject.h b/Include/methodobject.h
index 452f891a7a..39272815b1 100644
--- a/Include/methodobject.h
+++ b/Include/methodobject.h
@@ -31,7 +31,7 @@ typedef PyObject *(*PyCMethod)(PyObject *, PyTypeObject *, PyObject *const *,
 // Note that the underscore-prefixed names were documented in public docs;
 // people may be using them.
 typedef PyCFunctionFast _PyCFunctionFast;
-typedef PyCFunctionWithKeywords _PyCFunctionWithKeywords;
+typedef PyCFunctionFastWithKeywords _PyCFunctionFastWithKeywords;
 
 // Cast an function to the PyCFunction type to use it with PyMethodDef.
 //
diff --git a/Misc/NEWS.d/next/C API/2024-02-16-15-56-53.gh-issue-114626.ie2esA.rst b/Misc/NEWS.d/next/C API/2024-02-16-15-56-53.gh-issue-114626.ie2esA.rst
new file mode 100644
index 0000000000..763f4cee6d
--- /dev/null
+++ b/Misc/NEWS.d/next/C API/2024-02-16-15-56-53.gh-issue-114626.ie2esA.rst	
@@ -0,0 +1,4 @@
+Add again ``_PyCFunctionFastWithKeywords`` name, removed in Python 3.13
+alpha 4 by mistake.  Keep the old private ``_PyCFunctionFastWithKeywords``
+name (Python 3.7) as an alias to the new public name
+``PyCFunctionFastWithKeywords`` (Python 3.13a4). Patch by Victor Stinner.
